### PR TITLE
Add utility to parse duration string into ms

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves # .

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a utility function to parse duration string into milliseconds

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

Requesting review of `README.md` before rolling out implementation.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Review by kgryte — APPROVED** [review]

**Comment by Planeshifter** [comment]
This package is now ready for review.

File: lib/node_modules/@stdlib/time/duration2ms/README.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+</section>
+
+<!-- /.usage -->
+
+<section class="notes">
+
+## Notes
+
+-   A duration string is a string containing a sequence of time units. A time unit is a non-negative integer followed by a unit identifier. The following unit identifiers are supported:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/README.md]:
```suggestion
-   A duration string is a string containing a sequence of time units. A time unit is a nonnegative integer followed by a unit identifier. The following unit identifiers are supported:
```

File: lib/node_modules/@stdlib/time/duration2ms/benchmark/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		'1s',
+		'5ms',
+		'1ms'
+	];
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		ms = duration2ms( values[ i % values.length ] );
+		if ( !isInteger( ms ) ) {
+			b.fail( 'should return an integer' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/benchmark/benchmark.js]:
```suggestion
		if ( ms !== ms ) {
			b.fail( 'should not return NaN' );
```

File: lib/node_modules/@stdlib/time/duration2ms/bin/cli [hunk 3]
```diff
@@ -0,0 +1,107 @@
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/bin/cli:2]: Missing shebang.

File: lib/node_modules/@stdlib/time/duration2ms/docs/repl.txt [hunk 4]
```diff
@@ -0,0 +1,41 @@
+
+{{alias}}( str )
+    Parses a duration string and returns a number of milliseconds.
+
+    A duration string is a string containing a sequence of time units. A time
+    unit is a non-negative integer followed by a unit identifier. The following
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/docs/repl.txt]:
```suggestion
    unit is a nonnegative integer followed by a unit identifier. The following
```

File: lib/node_modules/@stdlib/time/duration2ms/docs/repl.txt [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    A duration string is a string containing a sequence of time units. A time
+    unit is a nonnegative integer followed by a unit identifier. The following
+    unit identifiers are supported:
+
+    -   `d`: days
+    -   `h`: hours
+    -   `m`: minutes
+    -   `s`: seconds
+    -   `ms`: milliseconds
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/docs/repl.txt]:
```suggestion
    -  d: days
    -  h: hours
    -  m: minutes
    -  s: seconds
    -  ms: milliseconds
```

Update to follow `repl.txt` style guide.

File: lib/node_modules/@stdlib/time/duration2ms/docs/types/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Parses a duration string and returns a number of milliseconds.
+*
+* ## Notes
+*
+* -   A duration string is a string containing a sequence of time units. A time unit is a non-negative integer followed by a unit identifier. The following unit identifiers are supported:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/docs/types/index.d.ts]:
```suggestion
* -   A duration string is a string containing a sequence of time units. A time unit is a nonnegative integer followed by a unit identifier. The following unit identifiers are supported:
```

File: lib/node_modules/@stdlib/time/duration2ms/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+
+// MAIN //
+
+/**
+* Parses a duration string and returns a number of milliseconds.
+*
+* ## Notes
+*
+* -   A duration string is a string containing a sequence of time units. A time unit is a non-negative integer followed by a unit identifier. The following unit identifiers are supported:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/lib/main.js]:
```suggestion
* -   A duration string is a string containing a sequence of time units. A time unit is a nonnegative integer followed by a unit identifier. The following unit identifiers are supported:
```

File: lib/node_modules/@stdlib/time/duration2ms/package.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    "name": "The Stdlib Authors",
+    "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+  },
+  "contributors": [
+    {
+      "name": "The Stdlib Authors",
+      "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+    }
+  ],
+  "main": "./lib",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/package.json:19]: Missing binary.

File: lib/node_modules/@stdlib/time/duration2ms/test/test.cli.js [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+		} else {
+			expected = '1350\n2000\n';
+			t.strictEqual( stdout.toString(), expected, 'expected value' );
+			t.strictEqual( stderr.toString(), '', 'does not print to `stderr`' );
+		}
+		t.end();
+	}
+});
+
+tape( 'the command-line interface supports use as a standard stream (delimiter)', opts, function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/test/test.cli.js]: Not clear why this description is different than L215.

File: lib/node_modules/@stdlib/time/duration2ms/test/test.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof duration2ms, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function throws an error if not provided a string', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/test/test.js]: Should update to "if not provided a duration string" and update the test values to include non-compliant strings.

**Review by kgryte — CHANGES_REQUESTED** [review]
Some minor nitpicks, but overall looks good.

**Comment by kgryte** [comment]
The only other thing I'd recommend is changing the description to

```
Convert a duration string to milliseconds.
```

The language about "parsing a duration string" is not necessary and can be more succinctly subsumed as part of conversion.

File: lib/node_modules/@stdlib/time/duration2ms/test/test.js (hunk 10 above)
Comment (Planeshifter) [reply lib/node_modules/@stdlib/time/duration2ms/test/test.js]: Updated.

File: lib/node_modules/@stdlib/time/duration2ms/package.json (hunk 8 above)
Comment (Planeshifter) [reply lib/node_modules/@stdlib/time/duration2ms/package.json:19]: Thanks for catching! Added.

File: lib/node_modules/@stdlib/time/duration2ms/test/test.cli.js (hunk 9 above)
Comment (Planeshifter) [reply lib/node_modules/@stdlib/time/duration2ms/test/test.cli.js]: Changed to test for when a regular expression is supplied for `split`.

**Comment by Planeshifter** [comment]
All changed!

File: lib/node_modules/@stdlib/time/duration2ms/bin/cli (hunk 3 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/bin/cli:2]: Still missing shebang here.

File: lib/node_modules/@stdlib/time/duration2ms/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+*
+* @example
+* var ms = duration2ms( '1d2h3m4s5ms' );
+* // returns 93784005
+*/
+function duration2ms( str ) {
+	var duration;
+	var out;
+
+	if ( isDurationString( str ) === false ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/lib/main.js]:
```suggestion
	if ( !isDurationString( str ) ) {
```

Typically, we test by negation.

File: lib/node_modules/@stdlib/time/duration2ms/bin/cli (hunk 3 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/time/duration2ms/bin/cli:2]:
```suggestion
#!/usr/bin/env node

```

**Review by kgryte — APPROVED** [review]
LGTM. Will merge once all checks pass.

**Pull request merged**